src/cplscheme/MultiCouplingScheme.cpp

```diff
@@ -52,52 +44,26 @@ void MultiCouplingScheme::initialize(
                       << getAcceleration()->getDataIDs().size());
     getAcceleration()->initialize(_allData); // Reserve memory, initialize
   }
+}
 
-  requireAction(constants::actionWriteIterationCheckpoint());
-  initializeTXTWriters();
+void MultiCouplingScheme::initializeImplementation()
+{
+  PRECICE_ASSERT(isImplicitCouplingScheme(), "MultiCouplingScheme is always Implicit.");
 
   for (DataMap &dataMap : _sendDataVector) {
-    for (DataMap::value_type &pair : dataMap) {
-      if (pair.second->initialize) {
-        setHasToSendInitData(true);
-        break;
-      }
-    }
+    initializeSendingParticipants(dataMap);
```

Comment:
That name is not good. What about "lookUpDataInitialization"? or sth similar.

Reply:
`initializeSendingParticipants` goes hand-in-hand with `initializeReceivingParticipants` (see [here](https://github.com/BenjaminRueth/precice/blob/c29a6e33a88d34b02b605c96d121950ce5a3e0e8/src/cplscheme/BaseCouplingScheme.hpp#L370-L374)). Therefore, we should probably change both names. I will try to understand better what the functions are doing, provide a documentation for the function and then update the names.

Reply:
I changed the names and added documentation in 8372e7c2e1ad0f74a1649d1a608d2e8be97de3d5. However, the names are pretty long. Something shorter would be nice, but I can also live with these names.

@uekerman How do you see the situation? If you are fine with the current state, we can mark this issue as resolved, from my point of view.

Reply:
@fsimonis and I came up with a nicer solution: 35219e5c71540566ac6157697206b448f7ed038e renames `CouplingData.initialize` to `requiresInitialization`. 334c5cda6f0514ef1b083bbcf182dd465e6f569f changes how the `bool` for data initialization is set. 1cf49e7b8349d25eb980966f79aa554c759e1fe7 changes the names of the member that control data initialization.

@uekerman Do you think this resolves the naming issue?

Reply:
Looks good!